Sensors/SMCBatteryManager/ACPIBattery.cpp

```diff
@@ -264,8 +276,9 @@ bool ACPIBattery::updateStaticStatus(bool *calculatedACAdapterConnection) {
 			IOSimpleLockUnlock(batteryInfoLock);
 			BatteryInfo bi;
 			bi.connected = true;
-			if (!getBatteryInfo(bi, true) && !getBatteryInfo(bi, false))
-				bi.connected = false;
+			if (!getBatteryInfo(bi, true))
+				if (!getBatteryInfo(bi, false))
```

Comment:
I think that's a leftover of some debugging, maybe return it back to `&&`?

Reply:
I'm not sure of this part. Or maybe keep the original format, but swap the sequence so _BIX can override _BIF?

Reply:
Well, currently the code does exactly the same thing, just is written strangely. As for BIX/BIF swapping, I am not sure this is true for all the models. Need @usr-sse2 here. Perhaps it could become a boot argument and/or a device property.

Reply:
We shouldn't call `_BIF` if `_BIX` succeeded, so try `_BIX` first and then `_BIF`, like it is. As `&&` operator is lazy, when `getBatteryInfo(bi, true)` returns `true`, the second operand is not evaluated. Please revert 016a763 as it doesn't change anything.